From 22f708a1ece613e011ec8740530b1ed40459891d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 22 Mar 2010 17:28:53 +0100
Subject: [PATCH 167/170] Revert "Fix crash on jack server shutdown"

This reverts commit e756467434f787d392c45aac1327768bcf0b610c.

The patch is incomplete and not entirely correct.
---
 src/modules/jack/module-jack-sink.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/jack/module-jack-sink.c b/src/modules/jack/module-jack-sink.c
index e4467d3..9f3e071 100644
--- a/src/modules/jack/module-jack-sink.c
+++ b/src/modules/jack/module-jack-sink.c
@@ -475,12 +475,12 @@ void pa__done(pa_module*m) {
     if (!(u = m->userdata))
         return;
 
-    if (u->sink)
-        pa_sink_unlink(u->sink);
-
     if (u->client)
         jack_client_close(u->client);
 
+    if (u->sink)
+        pa_sink_unlink(u->sink);
+
     if (u->thread) {
         pa_asyncmsgq_send(u->thread_mq.inq, NULL, PA_MESSAGE_SHUTDOWN, NULL, 0, NULL);
         pa_thread_free(u->thread);
-- 
1.7.1

